refaster-support/src/main/java/tech/picnic/errorprone/refaster/matchers/IsLambdaExpressionOrMethodReference.java

```diff
@@ -0,0 +1,39 @@
+package tech.picnic.errorprone.refaster.matchers;
+
+import static com.google.errorprone.matchers.Matchers.anyOf;
+import static com.sun.source.tree.MemberReferenceTree.ReferenceMode.INVOKE;
+
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.matchers.Matcher;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.LambdaExpressionTree;
+import com.sun.source.tree.MemberReferenceTree;
+import com.sun.source.tree.Tree;
+
+/** A matcher of lambda expressions or method references. */
+public final class IsLambdaExpressionOrMethodReference implements Matcher<ExpressionTree> {
+  private static final long serialVersionUID = 1L;
+
+  private static final Matcher<ExpressionTree> DELEGATE =
```

Comment:
While indeed _usually_ we introduce such a `DELEGATE` constant for performance reasons, in this case there's no efficiency gain over inlining the logic.